### PR TITLE
Creating Top Navigation Bar with Logos

### PR DESCRIPTION
**Pull request by eiffelwong1**
## Description

What changes have been made?
- Created a responsive top navigation bar, tested to 320px (iPhone se) up to 4k screen, confirmed responsiveness, menu item will need a future JS file to hide and unhide.

What tickets does this PR associate with? (Link to ticket)
https://github.com/social-plan-it/plan-it-social-web/issues/2

How has the UI changed? (Screenshot or recording, before or after)

phone - small
image

PC = 4K
image
note, the hero page is under another PR: create-hero-page (#9)

Does it work on mobile? (Screenshot for different screen sizes)
yes, see UI change

Has the database schema changed? If so, is it backward compatible?
N/A

Why were these changes made?
MVP

What tests have been added?
N/A, no front end test infra completed

How was it implemented, and why was this implementation chosen?

If a new npm package was added, why was it chosen and why was it necessary?
tailwindcss, we've decided to use

What alternatives were considered, and why weren't they used?
N/A

Documentation of challenges that were faced and how they were fixed
tailwindcss has a bit of an issue setting up, use `unstable_tailwind: true` in remix.config.ts instead of the recommended "tailwind: true,".

### CODE REVIEW

File: remix.config.js [hunk 1]
```diff
@@ -14,5 +14,6 @@ module.exports = {
     v2_meta: true,
     v2_normalizeFormMethod: true,
     v2_routeConvention: true,
+    unstable_tailwind: true,
```
Comment (ShowMale) [inline remix.config.js:17]: You may consider to add in 
module.exports = {
  postcss: true,
  tailwind: true,
  // ...
It worked in my project.

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "@types/react-dom": "^18.0.11",
     "eslint": "^8.38.0",
     "remix": "^1.16.0",
+    "tailwindcss": "^3.3.2",
     "typescript": "^4.9.5"
   },
```
Comment (ShowMale) [inline package.json:28]: Should we add "autoprefixer" dependency?

File: app/root.tsx [hunk 3]
```diff
@@ -14,7 +19,8 @@ import globalStyles from "./styles/global.css"
 import TopNav from "~/components/topNav";
 
 export const links: LinksFunction = () => [
-  { rel: "stylesheet", href: globalStyles },
+  // { rel: "stylesheet", href: globalStyles },
```
Comment (ShowMale) [inline app/root.tsx:22]: Probably, we can create styles/root.css file and import it here.

Comment (eiffelwong1) [reply app/root.tsx:22]: I should remove the commented line, the line below is the auto-generated tailwindcss related file, which is needed to use tailwind.

File: remix.config.js (hunk 1 above)
Comment (eiffelwong1) [reply remix.config.js:17]: we have tested tailwind:true, but we've noticed tailwind don't work with only "tailwind: true," will experiment with adding "postcss: true"

Comment (ShowMale) [reply remix.config.js:17]: When I pulled this branch and added "tailwind:true", it worked for me.

File: app/root.tsx (hunk 3 above)
Comment (ShowMale) [reply app/root.tsx:22]: Yeah, definitely you can delete it. I mean if we have some global CSS settings, we can define them in a separate root.css file.

File: package.json (hunk 2 above)
Comment (ShowMale) [reply package.json:28]: I mean if we are using "autoprefixer" and calling "npm install -D autoprefixer", it should be added to package.json as "autoprefixer": "^10.4.14".

**Review by andrelandgraf — COMMENTED** [review]
Great work! Let's merge #10 first and address the formatting issues before merging though!

File: app/root.tsx [hunk 4]
```diff
@@ -7,6 +12,17 @@ import {
   ScrollRestoration,
 } from "@remix-run/react";
 
+import globalStyles from "./styles/global.css"
```
[the remark this answers is not in the document]
Comment (eiffelwong1) [reply app/root.tsx:15]: will edit

File: app/root.tsx (hunk 3 above)
Comment (eiffelwong1) [reply app/root.tsx:22]: will remove

File: app/components/topNav.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
+import logo_hori from "../imgs/SocialPlan-it-logo-Horizontal.png";
+import logo_sq from "../imgs/SocialPlan-it-logo-Square.png";
+
```
Comment (ShowMale) [inline app/components/topNav.tsx:3]: Why don't move all images to the public folder?

File: app/components/topNav.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
+import logo_hori from "../imgs/SocialPlan-it-logo-Horizontal.png";
+import logo_sq from "../imgs/SocialPlan-it-logo-Square.png";
```
Comment (andrelandgraf) [inline app/components/topNav.tsx:2]: We have to talk about this 👀

File: app/components/topNav.tsx (hunk 5 above)
Comment (andrelandgraf) [reply app/components/topNav.tsx:3]: Great first move! What's the most important thing to consider with images and in what issues could we run when hosting our images as part of our code in the public folder?

File: package.json [hunk 7]
```diff
@@ -21,7 +21,9 @@
     "@remix-run/serve": "^1.7.2",
     "@types/react": "^18.0.35",
     "@types/react-dom": "^18.0.11",
-    "eslint": "^8.38.0",
+    "eslint": "^8.40.0",
+    "prettier": "2.8.8",
```
Comment (andrelandgraf) [inline package.json:25]: Wupsi, I don't think you wanted to add `eslint` and `prettier` here?

**Review by andrelandgraf — COMMENTED** [review]
You might want to merge the latest master into this branch. Other than that, ready to go!

**Comment by CamilingJS** [comment]
Can we simply the design to following?
image
I think a darker shade of blue/gray works better for the background color: "bg-gray-800"

**Review by andrelandgraf — DISMISSED** [review]
LGTM!

**Review by CamilingJS — DISMISSED** [review]
Looks good!

**Review by CamilingJS — APPROVED** [review]
Nice!

**Review by andrelandgraf — APPROVED** [review]

**Pull request merged**